src/dispatch/plugins/dispatch_slack/case/messages.py

```diff
@@ -511,15 +516,15 @@ def create_signal_engagement_message(
         organization_slug=case.project.organization.slug,
         project_id=case.project.id,
         channel_id=channel_id,
-        signal_instance_id=str(signal_instance.id),
-        engagement_id=engagement.id,
+        signal_instance_id=str(signal_instance.id) if signal_instance else "",
+        engagement_id=engagement.id if engagement else 0,
```

Comment:
Based on the type annotations `signal_instance` and  `engagement ` cannot be `None`, should the type annotations be updated?

Reply:
Good point. Updated.